controllers/nginx_controller.go

```diff
@@ -149,23 +152,27 @@ func (r *NginxReconciler) reconcileDeployment(ctx context.Context, nginx *nginxv
 }
 
 func (r *NginxReconciler) reconcileService(ctx context.Context, nginx *nginxv1alpha1.Nginx) error {
-	svcName := types.NamespacedName{
-		Name:      fmt.Sprintf("%s-service", nginx.Name),
-		Namespace: nginx.Namespace,
-	}
-
-	logger := r.Log.WithName("reconcileService").WithValues("Service", svcName)
-	logger.V(4).Info("Getting Service resource")
-
 	newService := k8s.NewService(nginx)
 
 	var currentService corev1.Service
-	err := r.Client.Get(ctx, svcName, &currentService)
-	if err != nil && errors.IsNotFound(err) {
-		logger.
-			WithValues("ServiceResource", newService).V(4).Info("Creating a Service resource")
+	err := r.Client.Get(ctx, types.NamespacedName{Name: newService.Name, Namespace: newService.Namespace}, &currentService)
+
+	if errors.IsNotFound(err) {
+		err = r.Client.Create(ctx, newService)
+		switch err {
```

Comment:
why not using classic if-else ?

Reply:
Just for personal preference :grimacing:

Reply:
Is it wordy?

Reply:
I though the same as @wpjunior 😅 
As there is only 1 switch and 1 default, I would go for a if-else

Reply:
if-else is simpler than switch.

Reply:
I did it to ease future mappings events from errors. However for now as we've a single mapping... it really doesn't do much sense.

Reply:
Did it